### PR TITLE
mongodb: fix 3.4 evaluation, mark others < 4.2 as broken

### PR DESCRIPTION
**Pull request by Emantor**
###### Motivation for this change
MongoDB was broken by the removal of pyyaml for python2. However it turns out that version 3.4 does not need python2 in the buildInputs at all, so remove it there and mark the other versions which depend on python2 and the pyyaml module as broken.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 1]
```diff
@@ -8,6 +8,7 @@ with lib;
 
 { version, sha256, patches ? []
 , license ? lib.licenses.sspl
+, broken ? false
```
Comment (bryanasdev000) [inline pkgs/servers/nosql/mongodb/mongodb.nix:11]: Can we use something else to mark as deprecated, per https://www.mongodb.com/support-policy/lifecycles ?

File: pkgs/servers/nosql/mongodb/v4_0.nix [hunk 2]
```diff
@@ -14,4 +14,5 @@ in buildMongoDB {
   patches =
     [ ./forget-build-dependencies.patch ./mozjs-45_fix-3-byte-opcode.patch ]
     ++ lib.optionals stdenv.isDarwin [ ./asio-no-experimental-string-view.patch ];
+  broken = true;
```
Comment (bryanasdev000) [inline pkgs/servers/nosql/mongodb/v4_0.nix:17]: I would leave v4.0, per https://www.mongodb.com/support-policy/lifecycles is not yet EOL'ed.

**Review by bryanasdev000 — CHANGES_REQUESTED** [review]

**Review by bryanasdev000 — CHANGES_REQUESTED** [review]
Besides my comments, can you please add a note to changelog? For the next release.

**Comment by Emantor** [comment]
This is out of date either way, so I'll close this. The programs built since pyyaml < 6.0 has been readded for python2.

**Pull request closed without merging**